File: soma_v2.cs
/******************************************************************************
Segunda versão do programa soma
*******************************************************************************/
using System;
class Operacao {
  static void Main() {
    
    //atribuição de dois valores as variaveis a e b
    //Int32.Parse converte para inteiro
    int a = Int32.Parse(Console.ReadLine());//captura do teclado o valor digitado
    int b = Int32.Parse(Console.ReadLine());//captura do teclado o valor digitado
    
    //chamo a função e ao mesmo tempo passo as variaveis
      Console.WriteLine ("Resultado foi: "+soma(a,b));
  }
  
  static int soma (int a, int b){
      int resultado = a + b;
      return resultado;
  }
}

File: soma_v2.java
/*segunda versão da função soma*/
import java.util.Scanner;//importação biblioteca de digitação
public class Main
{
	public static void main(String[] args) {
		
	    //Lê a partir da linha de comando
        Scanner sc = new Scanner(System.in);
		
		int a = sc.nextInt();
		int b = sc.nextInt();
		
		System.out.print("O resultado foi: "+ soma (a,b)); //chamo a função soma ao mesmo tempo passando os valores como paramentro

	}
	
	public static int soma (int a, int b){
	    int c = 0; // crio uma variavel local para receber o valor
	    c = a + b; // realizo a operação de soma
	    return c; // retorno o resultado
	}
}